### PR TITLE
Bug fix/check document key validity

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

Assertion thrown in stack below if an empty key is used. Handle it accordingly, produce proper error code. 

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [x] Backports required for: 

#### Related Information

Crash occured first with setting Replication to `2`

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket number: https://arangodb.atlassian.net/browse/BTS-253
- [ ] Design document: 

### Testing & Verification

This was discovered during improving the testing.

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [x] This change is already covered by existing tests, such as `shell-document.js` would trigger this missbehaviour.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [x] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/13139/

**Review by jsteemann — APPROVED** [review]

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13168/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13579/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13587/

File: js/client/modules/@arangodb/testutils/process-utils.js [hunk 1]
```diff
@@ -1760,6 +1760,7 @@ function startInstanceCluster (instanceInfo, protocol, options,
       coordinatorArgs['cluster.my-address'] = endpoint;
       coordinatorArgs['cluster.my-role'] = 'COORDINATOR';
       coordinatorArgs['cluster.agency-endpoint'] = agencyEndpoint;
+      coordinatorArgs['cluster.default-replication-factor'] = '2';
```
Comment (jsteemann) [inline js/client/modules/@arangodb/testutils/process-utils.js:1763]:
```suggestion
      // coordinatorArgs['cluster.default-replication-factor'] = '2'; // TODO: active later
```

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13593/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**